cmake/MOAB_PullAndMake.cmake

```diff
@@ -0,0 +1,44 @@
+# this Macro sets up the download and build of MOAB using ExternalProject
+# few tweak are done in src/dagmc/CMakeLists.txt and src/PyNE/CMakelists.txt 
+# to make sure that MOAB is built before DAGMC.
+MACRO (moab_pull_make moab_version)
+    message(STATUS "MOAB will be downloaded and built")
+    include(ExternalProject)
+    message("HDF5_ROOT: ${HDF5_ROOT}")
+    SET(MOAB_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}/moab")
+    set(MOAB_ROOT "${CMAKE_BINARY_DIR}/moab")
+    set(MOAB_INCLUDE_DIRS "${MOAB_INSTALL_PREFIX}/include")
+    set(MOAB_LIBRARY_DIRS "${MOAB_INSTALL_PREFIX}/lib")
+    message("MOAB_LIBRARY_DIRS: ${MOAB_LIBRARY_DIRS}")
+    set(MOAB_LIBRARIES_SHARED "")
+    ExternalProject_Add(MOAB_ep
+      PREFIX ${MOAB_ROOT}   
+      GIT_REPOSITORY https://bitbucket.org/fathomteam/moab.git
+      GIT_TAG ${moab_version}
+      CMAKE_ARGS
+        -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
+        -DBUILD_SHARED_LIBS:BOOL=ON
+        -DENABLE_HDF5:BOOL=ON
+        -DHDF5_ROOT:PATH=${HDF5_ROOT}
+        -DCMAKE_INSTALL_RPATH=${HDF5_ROOT}/lib:${MOAB_INSTALL_PREFIX}/lib
```

Comment:
This method may work here for now, but may create trouble while working with scikit-build-core depended project. However, I am not sure here.

Reply:
I have no experience about `scikit-build-core`, maybe we shall fix it, if the problem occurs ?

Reply:
have a doubt about what I did, I'll try one more time